src/utils/object.js

```diff
@@ -44,3 +44,41 @@ export function map(opt_initial) {
 export function hasOwn(obj, key) {
   return hasOwn_.call(obj, key);
 }
+
+/**
+ * Returns the value of the property referenced by dot-separated keys.
+ * e.g.
+ * `getPath('a.b[0].c', {a: {b: [{c: 2}]}}) === 2`
+ *
+ * @param {T} obj a map-like value
+ * @param {string} path a dot-separated list of keys to reference a value
+ * @return {T}
+ * @template T
+ */
+export function getPath(path, obj) {
+  const arrayIndexRe = /\[(\d+)\]/g;
+  const keys = path.replace(arrayIndexRe, '.$1').split('.');
+  return keys.reduce((acc, key) => acc[key], obj);
```

Comment:
CC @choumx who is doing this kind of stuff for amp-bind.

Since JSON returns full objects we need to guard this with `hasOwnProperty` to avoid exposing stuff like `constructor` and `toString`.

Please also think about how to make the error message good if this fails to find an element.